packages/istanbul-reports/lib/clover/index.js

```diff
@@ -135,13 +135,35 @@ class CloverReport extends ReportBase {
                 count,
                 type: 'stmt'
             };
-            const branchDetail = branchByLine[k];
+            const branchDetail = branchDetails[k];
 
-            if (branchDetail) {
-                attrs.type = 'cond';
-                attrs.truecount = branchDetail.covered;
-                attrs.falsecount = branchDetail.total - branchDetail.covered;
+            if (!branchDetail || branchDetail.type === 'switch') {
+                return this.xml.inlineTag('line', attrs);
             }
+
+            attrs.type = 'cond';
+            attrs.truecount = 0;
+            attrs.falsecount = 0;
+
+            if (count === 0) {
+                return this.xml.inlineTag('line', attrs);
+            }
+
+            if (['if', 'cond-expr'].includes(branchDetail.type)) {
```

Comment:
This logic is complex enough, that I think it would be good to add a description above each condition, explaining the reasoning for the logic.